src/Microsoft.OData.Core/UriParser/ODataUriParserSettings.cs

```diff
@@ -89,6 +89,19 @@ public ODataUriParserSettings()
             this.EnableParsingKeyAsSegment = true;
         }
 
+        internal ODataUriParserSettings Clone()
+        {
+            ODataUriParserSettings copy = new ODataUriParserSettings();
+            copy.FilterLimit = this.FilterLimit;
+            copy.OrderByLimit = this.OrderByLimit;
+            copy.PathLimit = this.PathLimit;
+            copy.SearchLimit = this.SearchLimit;
+            copy.MaximumExpansionDepth = this.MaximumExpansionDepth;
+            copy.MaximumExpansionCount = this.MaximumExpansionCount;
+            copy.EnableParsingKeyAsSegment = this.EnableParsingKeyAsSegment;
+            return copy;

```

Comment:
```suggestion
          return new ODataUriParserSettings()
           {
            FilterLimit = this.FilterLimit;
            OrderByLimit = this.OrderByLimit;
            PathLimit = this.PathLimit;
            SearchLimit = this.SearchLimit;
            MaximumExpansionDepth = this.MaximumExpansionDepth;
            MaximumExpansionCount = this.MaximumExpansionCount;
            EnableParsingKeyAsSegment = this.EnableParsingKeyAsSegment;
           };

```
nit: we could use theObject initializer here to make things clearer. Also not sure if we could use => new .... instead